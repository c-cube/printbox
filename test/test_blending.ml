let b =
  let open PrintBox in
  tree (frame @@ text "root") [
    frame @@ text "child 1";
    text "child 2";
    frame @@ tree empty [
      tree (frame @@ text "header 3") [frame @@ text "subchild 3"]
    ];
    tree empty [
      tree (frame @@ text "header 4") [frame @@ text "subchild 4"]
    ];
    frame @@ text "child 5";
    frame @@ tree (frame @@ text "header 6") [
      tree (frame @@ text "child 6") [frame @@ text "subchild 6"]
    ]
  ]

let () = print_endline @@ PrintBox_text.to_string b
