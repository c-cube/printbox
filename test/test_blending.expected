┌────┐
│root│
├────┘
├─┬───────┐
│ │child 1│
│ └───────┘
├─child 2
├─┬──────────────────┐
│ │──┬────────┐      │
│ │  │header 3│      │
│ │  ├────────┘      │
│ │  └─┬──────────┐  │
│ │    │subchild 3│  │
│ │    └──────────┘  │
│ └──────────────────┘
├───┬────────┐
│   │header 4│
│   ├────────┘
│   └─┬──────────┐
│     │subchild 4│
│     └──────────┘
├─┬───────┐
│ │child 5│
│ └───────┘
└─┬──────────────────┐
  │┌────────┐        │
  ││header 6│        │
  │├────────┘        │
  │└─┬───────┐       │
  │  │child 6│       │
  │  ├───────┘       │
  │  └─┬──────────┐  │
  │    │subchild 6│  │
  │    └──────────┘  │
  └──────────────────┘
